### ISSUE TITLE
Windows library requires Visual C++ redistributable x86 and x64

### ISSUE THREAD

**Issue by janeshparnami** [asus4/tf-lite-unity-sample#244]
_I get the following error when running BlazePose._

**Failed to load 'C:/Users/janes/UnityProjects/Work/tf-lite-unity-sample-master/Packages/com.github.asus4.tflite/Plugins/Windows/libtensorflowlite_c.dll', expected x64 architecture, but was Unknown architecture. You must recompile your plugin for x64 architecture.**

_After Playing I get the above error multiple times then this new error once_

**DllNotFoundException: libtensorflowlite_c
TensorFlowLite.InterpreterOptions..ctor () (at Packages/com.github.asus4.tflite/Runtime/InterpreterOptions.cs:60)
TensorFlowLite.BaseImagePredictor`1[T]..ctor (System.String modelPath, TensorFlowLite.BaseImagePredictor`1+Accelerator[T] accelerator) (at Packages/com.github.asus4.tflite.common/Runtime/BaseImagePredictor.cs:56)
TensorFlowLite.PoseDetect..ctor (TensorFlowLite.PoseDetect+Options options) (at Assets/Samples/BlazePose/PoseDetect.cs:57)
TensorFlowLite.BlazePose..ctor (TensorFlowLite.BlazePose+Options options) (at Assets/Samples/BlazePose/BlazePose.cs:39)
BlazePoseSample.Start () (at Assets/Samples/BlazePose/BlazePoseSample.cs:43)**

_Then this error repeats continuously_

**NullReferenceException: Object reference not set to an instance of an object
BlazePoseSample.Update () (at Assets/Samples/BlazePose/BlazePoseSample.cs:76)**

I have Unity Hub 3.2.0 and UnityEditor 2020.3.36f1.

Thanks.

![BlazeposeErrorGithub](https://user-images.githubusercontent.com/11949264/186469424-d3596e0a-231d-47e4-9c22-29139501250d.png)

**Comment by asus4** [comment]
Just confirming, have you checked out with Git LFS? Please confirm the file size of `libtensorflowlite_c.dll`.  It should be around 2.8MB.

**Comment by janeshparnami** [comment]
Yes.
The file size is 2780Kb.
I have not checked with Git LFS. I am new to this, do not know what it is.
When I started the Project for the first time it asked for Git integration, so I installed Git checked the Issues Page and figured I have to add the Environment Path as well, so did that and then it opened in Unity but then all these  errors come.

I am using Windows 11.

**Comment by asus4** [comment]
@janeshparnami Thanks, the DLL seems to be correct. Could you describe your environment details? I have not tested this on Windows 11. 

Also, issue #39 might be related.

**Comment by janeshparnami** [comment]
Hi @asus4,
I am using **i5 9th generation with 4 Gb nvidia gtx 1650** . Using it mostly for entertainment purposes and Unity. Only big software I can think of is Blender.

Issue #39 actually solved half my problem.
https://github.com/asus4/tf-lite-unity-sample/issues/39#issuecomment-719993074

I installed Visual C++ redistributable x86 and x64 latest versions. It did not work on the already open project even after restaring unity and even PC.

So i redownloaded the code zip again and it **started working in Unity.**

It solves half my problem because my colleague and I were testing it by detecting a jump.Since I was not able to run Blazepose on my PC, I was doing the Unity stuff and he was working with the code in this repository.

That project still not works on my PC along with the version I opened before installing Visual C++ redistributable.
I guess I will have to redo the work again on the new empty tfliteunitySample that works for me.


If you have any suggestion as to how I can make the projects already opened before Visual C++ installed, to get to work, it would be very helpful.

Thanks for all the help though, it finally works on my PC atleast. 👍
